### PR TITLE
Adds map generators to build mode

### PR DESCRIPTION
**Pull request by AnturK**
space station 13 2015-05-21 192700
- Adds generate area mode to build mode.
- Adds few variations of asteroid template
- Adds syndicate template
- Some minor changes to build mode.

Added to make preparing areas for events less time consuming.

Not sure if admin tools count for freeze but anyway this is not finished yet so DNM please.

### CODE REVIEW

**Comment by RemieRichards** [comment]
Wonderful use of the system, I approve :+1:

File: code/modules/admin/verbs/buildmode.dm [hunk 1]
```diff
@@ -174,6 +179,12 @@
 						master.buildmode.valueholder = input(usr,"Enter variable value:" ,"Value") as obj in world
 					if("turf-reference")
 						master.buildmode.valueholder = input(usr,"Enter variable value:" ,"Value") as turf in world
+			if(AREA_BUILDMODE)
+				var/list/gen_paths = typesof(/datum/mapGenerator) - /datum/mapGenerator
+
+				var/type = input(usr,"Select Generator Type","Type") in gen_paths
```
Comment (RemieRichards) [inline code/modules/admin/verbs/buildmode.dm]: ``` as null|anything in gen_paths```
and then a check for ```if(!type) return ``` so we can quickly cancel.

**Comment by Dennok** [comment]
Wonderful

**Comment by Stickymayhem** [comment]
Could you add a shuttle one as well that uses simulated shuttle turfs and walls that are properly iconed?

Even a simple square outline would save so much fucking time in shuttle events.

**Comment by AnturK** [comment]
Sure, I'll be finishing this tommorow will do that then.

**Comment by AnturK** [comment]
This is finished for now. I'll add custom/circular shape areas later.

**Comment by optimumtact** [comment]
@RemieRichards merge?

**Pull request merged**